assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -393,61 +273,48 @@ pub fn u32rotr(
     op_mode: U32OpMode,
     imm: Option<u8>,
 ) -> Result<Option<CodeBlock>, AssemblyError> {
-    match imm {
-        None => match op_mode {
-            U32OpMode::Checked => {
+    match (imm, op_mode) {
+        (Some(imm), U32OpMode::Checked) => {
+            // Assert the top of the stack is a u32 value.
+            // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
+            // than u32 for b.
+            // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
+            // guaranteed to be u32 values
```

Comment:
Let's create an issue to double-check this and the whole function for `u32rotr` and maybe add more tests. Something here doesn't look right to me.

Reply:
https://github.com/maticnetwork/miden/issues/541